src/Microsoft.OData.Core/ODataCollectionWriterCore.cs

```diff
@@ -178,10 +178,11 @@ public sealed override void WriteStart(ODataCollectionStart collectionStart)
         /// </summary>
         /// <param name="collection">The <see cref="ODataCollectionStart"/> representing the collection.</param>
         /// <returns>A task instance that represents the asynchronous write operation.</returns>
-        public sealed override Task WriteStartAsync(ODataCollectionStart collection)
+        public sealed override async Task WriteStartAsync(ODataCollectionStart collection)
         {
             this.VerifyCanWriteStart(false, collection);
-            return TaskUtils.GetTaskForSynchronousOperation(() => this.WriteStartImplementation(collection));
+            await this.WriteStartImplementationAsync(collection)

```

Comment:
cant we elide this?

Reply:
It was being done before but its actually not a safe thing to do here. `VerifyCanWriteStart` can throw an exception and if this method is not async, an AsyncStateMachine will not be created, meaning that the exception would escape to the caller(s).